plugins/modules/vmware_guest_controller.py

```diff
@@ -410,6 +422,11 @@ def sanitize_disk_controller_config(self):
                             ctl_config['controller_number'] = self.disk_ctl_bus_num_list['scsi'].pop(0)
                         else:
                             ctl_config['controller_number'] = None
+
+                        # Set default value for bus_sharing if not given
+                        if ctl_config.get('bus_sharing') is None:
+                            ctl_config['bus_sharing'] = 'noSharing'
```

Comment:
I would remove this, it's not needed since ansible will set `bus_sharing` to `noSharing` as defined in `argument_spec`.